### PR TITLE
Added tests for sign-in and sign-up screens

### PR DESCRIPTION
**Pull request by mahdiatallah**
This PR introduces comprehensive tests for the Sign-In and Sign-Up screens, ensuring at least 80% coverage. The tests are updated to reflect recent changes in the app's structure and navigation following M2, providing robust validation for the revised functionality.

### CODE REVIEW

File: app/__tests__/SignInScreen.test.tsx
Comment (baudoin-csp) [inline app/__tests__/SignInScreen.test.tsx:1]: Great job overall! I think there is a small typo in the import that you can fix :)

See: ```import SignInScreen from '../screens//auth/SignInScreen';```

File: app/screens/auth/SignInScreen.tsx
Comment (baudoin-csp) [inline app/screens/auth/SignInScreen.tsx:1]: You correctly removed aliases to make the tests work! Consider removing the unused imports such as: ```GoogleSignInButton```, ```useAuth``` and the unnecessary comments.

I am not familiar with the design of the screen but may I ask you if it's possible to reuse more styles from ```styles/styles.txt``` if it's not possible.

File: sonar-project.properties
Comment (baudoin-csp) [inline sonar-project.properties:1]: That's a good fix thank you! Finally the tests are not part of the coverage anymore 👍

**Review by baudoin-csp — COMMENTED** [review]
That's a very good PR @mahdiatallah! The length is perfect and it's easy to understand and to review. You correctly applied the policy that we discussed during the last standup meeting, meaning: delete all the not working tests to always pass the CI when merging on main. 
All the unit tests for the SignInScreen passed and the implementation of the screen is correct. Kudo for that 👍
Screenshot 2024-11-22 at 09 17 06
Consider adding the small fixes mentioned to merge this PR into main.

File: app/screens/auth/SignUpScreen.tsx
Comment (baudoin-csp) [inline app/screens/auth/SignUpScreen.tsx:1]: Consider removing unused imports: ```emailAndPasswordSignIn, Alert, createTenant, createUser```.

Comment (baudoin-csp) [reply app/screens/auth/SignUpScreen.tsx:1]: The interface ```FormErrors```is not used also.

**Comment by mahdiatallah** [comment]
Thanks for your insights. I addressed the changes and pulled the latest changes from main. Let me know if there's anything else to improve! 🚀

**Comment by baudoin-csp** [comment]
That's all good! Thank you @mahdiatallah

**Review by baudoin-csp — COMMENTED** [review]
Well done for your job @mahdiatallah !

**Pull request merged**